File: app/src/main/java/com/example/haushaltsapp/types/Supply.java
package com.example.haushaltsapp.types;

import androidx.annotation.NonNull;

import java.util.List;

class Supply {

    @NonNull
    private List<SupplyEntry> entries;

    @NonNull
    private User owner;

    @NonNull
    private List<User> members;

    public Supply(@NonNull List<SupplyEntry> entries, @NonNull User owner, @NonNull List<User> members) {
        this.entries = entries;
        this.owner = owner;
        this.members = members;
    }

    @NonNull
    public List<SupplyEntry> getEntries() {
        return entries;
    }

    public void setEntries(@NonNull List<SupplyEntry> entries) {
        this.entries = entries;
    }

    @NonNull
    public User getOwner() {
        return owner;
    }

    public void setOwner(@NonNull User owner) {
        this.owner = owner;
    }

    @NonNull
    public List<User> getMembers() {
        return members;
    }

    public void setMembers(@NonNull List<User> members) {
        this.members = members;
    }
}

File: app/src/main/java/com/example/haushaltsapp/types/Supply.kt
package com.example.haushaltsapp.types

data class Supply(
        var id: String,
        var name: String,
        val entries: MutableList<SupplyEntry>,
        val owner: UserSummary,
        val members: MutableList<UserSummary>
) {
    // @Suppress("unused")
    //constructor() : this(0, mutableListOf(), UserSummary(), mutableListOf())

    @Suppress("unused")
    constructor() : this("", "", mutableListOf(), UserSummary(), mutableListOf())

    constructor(name: String, owner: UserSummary) : this("", name, mutableListOf(), owner, mutableListOf())

    fun toSummary(): ShoppingListSummary = ShoppingListSummary(id, name)

}
